src/stream.rs

```diff
@@ -196,8 +203,22 @@ impl Compress {
     /// Creates a new compressor ready to encode data into brotli
     pub fn new(params: &CompressParams) -> Compress {
         unsafe {
-            let state = brotli_sys::RustBrotliCompressorCreate(params.params);
+            let state = brotli_sys::BrotliEncoderCreateInstance(None, None, 0 as *mut _);
             assert!(!state.is_null());
+
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::BROTLI_PARAM_MODE,
+                                                  params.mode_as_native() as uint32_t);
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::BROTLI_PARAM_QUALITY,
+                                                  params.quality);
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::BROTLI_PARAM_LGWIN,
+                                                  params.lgwin);
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::BROTLI_PARAM_LGBLOCK,
+                                                  params.lgblock);
```

Comment:
Actually if we remove `compress_vec` below then we can jettison `CompressParams` entirely and just make these methods on the `Compress` structure itself. I'll just do that locally though, the rest of this looks good to go.